File: aoc2015/day1/solution.py
"""Day 1"""

def get_floor(instructions: str) -> int:
    """Calculation one instruction."""

    floor = 0

    for instruction in instructions:
        if instruction == '(':
            floor += 1
        elif instruction == ')':
            floor -= 1
    return floor

def get_basement(instructions: str) -> int:
    """Calculation first enter the basement."""

    floor = 0
    position = 0
    i = 0

    for instruction in instructions:
        if instruction == '(':
            floor += 1
        elif instruction == ')':
            floor -= 1
        position += 1
        if floor == -1 and i == 0:
            i += 1
            return position

def main():
    """Main."""
    filename = '/home/goddriver/Documents/exercises/aoc2015/1/input'

    with open(filename) as f_i:
        elements = f_i.read()

    print(f"In floor {get_floor(elements)}.")
    print(f"First enter the basement is at character position \
{get_basement(elements)}.")

if __name__ == "__main__":
    main()

File: aoc2015/day1/solution.java
package ru.goddriver.exercises.aoc2015.day1;

import java.io.BufferedReader;
import java.io.FileReader;
import java.util.Scanner;

public class solution {

    private static int getFloor(String fileName) throws Exception {
        Scanner scanner = new Scanner(new BufferedReader(new FileReader(fileName)));
        String instruction = scanner.nextLine();
        int floor = 0;
        for (int i = 0; i < instruction.length(); i++) {
            if (instruction.charAt(i) == '(')
                floor++;
            if (instruction.charAt(i) == ')')
                floor--;
        }
        scanner.close();
        return floor;
    }

    private static int getFirstEnterTheBasement(String fileName) throws Exception {
        Scanner scanner = new Scanner(new BufferedReader(new FileReader(fileName)));
        String instruction = scanner.nextLine();
        int floor = 0;
        while (true) {
            for (int i = 0; i < instruction.length(); i++) {
                if (instruction.charAt(i) == '(')
                    floor++;
                if (instruction.charAt(i) == ')')
                    floor--;
                if (floor == -1) {
                    scanner.close();
                    return i + 1;
                }
            }
        }
    }

    public static void main(String[] args) throws Exception{
        String fileName = "input";
        //Part One
        System.out.println("Floor at the Santa's instruction is " + getFloor(fileName));
        //Part Two
        System.out.println("First enter the basement on position " + getFirstEnterTheBasement(fileName));
    }
}